scripts/extend-queries.sh

```diff
@@ -22,21 +23,34 @@ invoke_contract() {
 
   echo "Calling function: $FUNCTION_NAME on contract: $CONTRACT_ID with args: $ARGS" >&2
 
-  stellar contract invoke \
+  # capture raw output and errors
+  RAW_OUTPUT=$(stellar contract invoke \
     --id $CONTRACT_ID \
     --source-account $SOURCE_ACCOUNT \
     --rpc-url $RPC_URL \
     --network-passphrase "$NETWORK_PASSPHRASE" \
     --send $SEND_TX \
     -- \
-    $FUNCTION_NAME $ARGS | jq
+    $FUNCTION_NAME $ARGS 2>&1)
+
+  # check for archived entry error
+  if echo "$RAW_OUTPUT" | grep -q "EntryArchived"; then
+    echo "Error: EntryArchived for function: $FUNCTION_NAME on contract: $CONTRACT_ID" >&2
+    return 1
+  fi
+
+  # parse JSON output with jq
+  echo "$RAW_OUTPUT" | jq
 }
 
+# redirect all output to the specified file or /dev/null
+exec >"$OUTPUT_FILE" 2>&1
+
 # all pools of factory
 POOLS=$(invoke_contract $FACTORY_ID query_pools | jq -r '.[]')
 
 # query the pools and get the details
-echo "Iterating over pool addresses"
+echo "Iterating over pool addresses" >&2
```

Comment:
Those messages are fine, I meant only the soroban cli outputs since they are not important.

Reply:
In one of the very first commits, I added the following line for logging purposes: 
```bash
echo "Calling function: $FUNCTION_NAME on contract: $CONTRACT_ID with args: $ARGS" >&2
```
 
The reason for redirecting this output to standard error (>&2) is to separate my logging messages from the output generated by jq. Without this separation, jq fails to produce a correct response, which disrupts the entire logging mechanism of the script.